reconcile/utils/gql.py

```diff
@@ -188,6 +188,22 @@ def get_resource(self, path: str) -> dict[str, Any]:
 
         return resources[0]
 
+    def get_resources_by_schema(self, schema: str) -> list[dict[str, str]]:
+        query = """
+        query Resource($schema: String) {
+            resources: resources_v1 (schema: $schema) {
+                path
+                content
+                sha256sum
+            }
+        }
+        """
+
+        # Do not validate schema in resources since schema support in the
+        # resources is not complete.
+        resources = self.query(query, {"schema": schema}, skip_validation=True)
```

Comment:
Not sure I understand this. How can we query sth with gql that is not defined in the schema?

Reply:
This is a very good question! Actually, I've copied and pasted the original [get_resource](https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/gql.py#L166) method and adapted it 🤪

Reply:
@rporres do you know hot this is possible?

Reply:
Or maybe @geoberle can explain that?

Reply:
Hmmm, there's no GraphQL schema for anything under `app-interface/resources`, only JSON schema. GraphQL schema only refers to objects under `data`. Resources are usually free form YAML objects and we only add validation in a few of them.